### ISSUE TITLE
`ormolu-0.5.0.1` fail to build with Nix …

### ISSUE THREAD

**Issue by yvan-sraka** [tweag/ormolu#965]
**Describe the bug**

`ormolu-0.5.0.1` fail to build with Nix, the expression is defined here: https://github.com/NixOS/nixpkgs/blob/nixos-22.11/pkgs/development/haskell-modules/hackage-packages.nix#L211319

**To Reproduce**

`nix-shell -p ormolu` with current `nixpkgs-unstable`, will produce error:

```
error: cycle detected in build of '/nix/store/mn4cj42d5b99i1iky0k6z0iq0m8jwrix-ormolu-0.5.0.1.drv' in the references of output 'bin' from output 'out'
```

The content of [`/nix/store/mn4cj42d5b99i1iky0k6z0iq0m8jwrix-ormolu-0.5.0.1.drv`](https://github.com/tweag/ormolu/files/10364151/mn4cj42d5b99i1iky0k6z0iq0m8jwrix-ormolu-0.5.0.1.drv.txt) produced.

**Expected behavior**

I expect it to build successfully, here my full  [build log](https://github.com/tweag/ormolu/files/10364202/build_log.txt).

**Environment**

 - OS name + version: I'm on `darwin-aarch64`
 - Version of the code: Trying to build `ormolu-0.5.0.1`

**Comment by amesgen** [comment]
I can't reproduce this on x86_64-linux, latest `nixpkgs-unstable` commit is 298add347c2bbce14020fcb54051f517c391196b right now, and both
```console
nix shell github:NixOS/nixpkgs/298add347c2bbce14020fcb54051f517c391196b#ormolu
nix-shell -I nixpkgs=https://github.com/NixOS/nixpkgs/archive/298add347c2bbce14020fcb54051f517c391196b.zip -p ormolu
```
drop me into a shell with a working Ormolu 0.5.0.1.

Either way, this seems to be packaging-related issue, so I think this can be closed in favor of 
your ticket https://github.com/NixOS/nixpkgs/issues/209397?

**Comment by yvan-sraka** [comment]
> ```
> nix shell github:NixOS/nixpkgs/298add347c2bbce14020fcb54051f517c391196b#ormolu
> nix-shell -I nixpkgs=https://github.com/NixOS/nixpkgs/archive/298add347c2bbce14020fcb54051f517c391196b.zip -p ormolu
> ```

I confirm that I could reproduce on my machine with both of the commands you suggest.

> Either way, this seems to be purely a packaging issue, so I think this can be closed in favor of your ticket [NixOS/nixpkgs#209397](https://github.com/NixOS/nixpkgs/issues/209397)?

Agreed with that.
